Index: mdkonline-2.78/mdkapplet
===================================================================
--- mdkonline-2.78.orig/mdkapplet
+++ mdkonline-2.78/mdkapplet
@@ -456,7 +456,7 @@ sub is_download_all_enabled() {
 sub confirm_upgrade() {
     local $mygtk2::left_padding = 0;
     my $warn_me = text2bool($local_config{DO_NOT_ASK_FOR_DISTRO_UPGRADE});
-    my $w = new_portable_dialog(N("New version of Mandriva Linux distribution"));
+    my $w = new_portable_dialog(N("New version of OpenMandriva Lx"));
     my ($temp_dir, $box);
     
     my $browse;
@@ -464,9 +464,7 @@ sub confirm_upgrade() {
         Gtk2::FileChooserButton->new(N("Browse"), 'select-folder'),
         'current-folder-changed' => sub {
             $temp_dir = $_[0]->get_current_folder;
-            my $ok = 1;
-            require urpm;
-            urpm::check_cache_dir({ fatal => sub { $ok = 0 } }, $temp_dir);
+            my $ok = -d $temp_dir && ! -l $temp_dir && ((stat($temp_dir))[4] == 0);
             $ok or ask_warn(N("Error"), N("You must choose a directory owned by the super administrator!"));
         });
     $browse->set_current_folder($download_dir);
@@ -474,34 +472,29 @@ sub confirm_upgrade() {
     my $res = fill_n_run_portable_dialog($w,
                                       [
                get_banner(),
-               gtknew('Label_Left', text => N("A new version of Mandriva Linux distribution has been released."), @common),
+               gtknew('Label_Left', text => N("A new version of OpenMandriva Lx has been released."), @common),
                gtknew('HButtonBox', layout => 'start', children_tight => [
                    new_link_button($new_distro->{url}, N("More info about this new version")),
                ]),
                gtknew('Label_Left', text => N("Do you want to upgrade to the '\%s' distribution?", 
                                               $new_distro->{name} || $new_distro->{version}), @common),
                gtknew('CheckButton', text => N("Do not ask me next time"), active_ref => \$warn_me),
-               (is_download_all_enabled() ?
-                   (gtknew('CheckButton', text => N("Download all packages at once") . "\n" .
-                            N("(Warning: You will need quite a lot of free space)"),
-                          active_ref => \$download_all, sensitive_ref => \$browse,
-                          toggled => sub { $box and $box->set_sensitive($download_all) },
-                      ),
-                   $box = gtknew('HBox', sensitive => $download_all,
-                                 children => [
-                                     0, gtknew('Label_Left', text => N("Where to download packages:")),
-                                     1 , $browse,
-                                 ]),
-               ) : ()),
+               gtknew('CheckButton', text => N("Download all packages at once") . "\n" .
+                   N("(Warning: You will need quite a lot of free space)"),
+                   active_ref => \$download_all, sensitive_ref => \$browse,
+                   toggled => sub { $box and $box->set_sensitive($download_all) },
+               ),
+               $box = gtknew('HBox', sensitive => $download_all, children => [
+                    0, gtknew('Label_Left', text => N("Where to download packages:")),
+                    1 , $browse,
+               ]),
                create_okcancel($w, N("Next"), N("Cancel")),
                                       ]);
 
     setVar('DO_NOT_ASK_FOR_DISTRO_UPGRADE', bool2text($warn_me));
     $local_config{DO_NOT_ASK_FOR_DISTRO_UPGRADE} = bool2text($warn_me);
     if ($res) {
-        require urpm;
-        my $ok = 1;
-        urpm::check_cache_dir({ fatal => sub { $ok = 0 } }, $temp_dir);
+        my $ok = -d $temp_dir && ! -l $temp_dir && ((stat($temp_dir))[4] == 0);
         $ok or goto &confirm_upgrade;
         $download_dir = $temp_dir;
         really_confirm_upgrade();
@@ -512,7 +505,7 @@ sub confirm_upgrade() {
 
 
 sub get_obsolete_message_() {
-    N("Maintenance for this Mandriva Linux version has ended. No more updates will be delivered for this system."),
+    N("Maintenance for this OpenMandriva Lx version has ended. No more updates will be delivered for this system."),
 }
 
 sub get_obsolete_message() {
@@ -524,7 +517,7 @@ sub get_obsolete_message() {
       
 sub refresh_no_more_supported_msg() {
     my $basic_msg = get_obsolete_message_();
-    my $distro = N("Mandriva Linux");
+    my $distro = N("OpenMandriva Lx");
     my $msg;
 
     # is extended maintenance available?
@@ -638,7 +631,7 @@ sub get_extended_maintenance() {
 
 sub really_confirm_upgrade() {
     local $mygtk2::left_padding = 0;
-    my $w = ugtk2->new(N("New version of Mandriva Linux distribution"), width => $width + 20);
+    my $w = ugtk2->new(N("New version of OpenMandriva Lx"), width => $width + 20);
 
     # estimated package size:
     my $c;
@@ -664,7 +657,7 @@ sub really_confirm_upgrade() {
                              N("You should put your laptop on AC and favor ethernet connection over wifi, if available.")
                                : ''),
                       @common),
-               create_okcancel($w, N("Next"), N("Cancel")),
+               create_okcancel($w, N("Start"), N("Cancel")),
            ]),
        );
     }
