--- mdkonline-2.78/mdkapplet.orig	2014-05-03 19:10:03.505799334 +0100
+++ mdkonline-2.78/mdkapplet	2014-05-03 19:10:17.788799912 +0100
@@ -466,7 +466,7 @@
             $temp_dir = $_[0]->get_current_folder;
             my $ok = 1;
             require urpm;
-            urpm::check_cache_dir({ fatal => sub { $ok = 0 } }, $temp_dir);
+            urpm::check_dir({ fatal => sub { $ok = 0 } }, $temp_dir);
             $ok or ask_warn(N("Error"), N("You must choose a directory owned by the super administrator!"));
         });
     $browse->set_current_folder($download_dir);
@@ -501,7 +501,7 @@
     if ($res) {
         require urpm;
         my $ok = 1;
-        urpm::check_cache_dir({ fatal => sub { $ok = 0 } }, $temp_dir);
+        urpm::check_dir({ fatal => sub { $ok = 0 } }, $temp_dir);
         $ok or goto &confirm_upgrade;
         $download_dir = $temp_dir;
         really_confirm_upgrade();
